consensus/abci.go

```diff
@@ -123,17 +122,50 @@ func (abci *TendermintABCI) CheckTx(txBytes []byte) tmtAbciTypes.ResponseCheckTx
 	
 	tx, err := decodeRLP(txBytes)
 	if err != nil {
-		abci.logger.Error("Received invalid transaction", "tx", tx.Hash().String())
-		abci.metrics.CheckErrTxsTotal.Add(1, fmt.Sprint(abciTypes.ErrEncodingError.Code))
-		return tmtAbciTypes.ResponseCheckTx{Code: uint32(abciTypes.ErrEncodingError.Code), Log: err.Error()}
+		abci.logger.Error("Unable to decode RLP TX", "err", err.Error())
+		abci.metrics.CheckErrTxsTotal.Add(1, "INVALID_TX")
+		return tmtAbciTypes.ResponseCheckTx{Code: 1, Log: "INVALID_TX"}
 	}
 
 	abci.logger.Info("Checking TX", "hash", tx.Hash().String(), "nonce", tx.Nonce(), "cost", tx.Cost())
+	from, err := abci.doMempoolValidation(tx)
+	if err != nil {
+		abci.logger.Error(err.Error())
+		abci.metrics.CheckErrTxsTotal.Add(1, err.Error())
+		return tmtAbciTypes.ResponseCheckTx{Code: 1, Log: err.Error()}
+	}
+
+	// Additional custom validation aside of the Ethereum default mempool validation
+	if tx.GasPrice().Uint64() < abci.db.Config().TxPool.PriceLimit {
```

Comment:
[QUESTION] Is this validation adding any additional security over the `IntrinsicGas` validation? 

[MINOR} What is the reason to have this validation as not part of the `doMempoolValidation`? As I see it, we are checking the gas twice in two manners and there is not sense to be splitting that validation into two different places, that will add confusion.

Reply:
gas amount != gas price, they are different concepts. IntrinsicGas defines the amount of gas necessary for an operation and is constant in case of funds transfers, gas price works like an auction system and is dynamic. Also this validation together with balances is custom and not part of the shared mempool validation